persistence/sql/persister_consent.go

```diff
@@ -438,3 +438,14 @@ func (p *Persister) VerifyAndInvalidateLogoutRequest(ctx context.Context, verifi
 		return nil
 	})
 }
+
+func (p *Persister) FlushInactiveLoginConsentRequests(ctx context.Context, notAfter time.Time) error {
+	var lr consent.LoginRequest
+	err := p.Connection(ctx).RawQuery(
+		fmt.Sprintf("DELETE FROM %s WHERE requested_at < ? AND requested_at < ?", (&lr).TableName()),
```

Comment:
I don't think this will do what you want - it removes stuff from the login request table which is older than the request max age. The problem however is that we want to keep this stuff around for successful flows - otherwise all tokens associated with this request will be invalidated! So we probably need to use some JOIN magic to achieve that!

Reply:
Oh i see - so this table also gets checked for the token validity?
So basically when removing from here - also check that the token has expired or has no session?

Reply:
You're actually right, I think it is ok to remove entries from this table. I thought it might have cascading effects but as far as I can tell it might not. The only cascading delete I could find was on [`hydra_oauth2_authentication_request_handled`](https://github.com/ory/hydra/blob/9a81fbcf804b199da0f02c9f07e1adf42d418927/internal/fizzmigrate/consent/migrations/sql/postgres/7.sql#L43) which *should* be ok.

This will however not work for the consent table as removing stuff from the consent table that is valid will break a lot of flows.

However, I think that this approach is still not the best one. Instead, you should delete only rows that have failed flows. This means flows which:

- Do not have a corresponding `*_handled` row which means that the flow never completed
- Have a `*_handled` flow which does not have the status of accepted